### ISSUE TITLE
Missing dwarf information from gcc 8.1.0 binary

### ISSUE THREAD

**Issue by mplegendre** [dyninst/dyninst#625]
Dyninst is reporting that a build of lulesh with gcc 8.1.0 doesn't have line information associated with lulesh.cc:233, and that function CollectDomainNodesToElemNodes doesn't have a local variable named nd1i.  Based on other DWARF parsers, I don't believe either of these are true.

I've tested this with Dyninst^540b228 and 10.0.  I'll send an email to @sashanicolas @hainest and @woodard with this lulesh binary, and a small Dyninst reproducer that does the two lookup operations.  

It's not clear to me what's the correct return for the line information.  GDB inserts a breakpoint at lulesh.cc:233 at 0x4060a0 and 0x409c0c.  dwarfdump -l shows addresses at 0x4060a0, 0x409c0c, and 0x403eb0.  addr2line and Dyninst report no addresses.  
Based on my read of the assembly, I think the line is implemented at 0x4060ab.

At the missing line lulesh.cc:233 is the declaration of the variable nd0i, which is also missing in Dyninst.  dwarfdump clearly shows that info for this variable exists.

**Comment by hainest** [comment]
@mxz297 could you give us your thoughts?

**Comment by mxz297** [comment]
It is difficult to predict what's the problem. But based on what Sasha has described that Dyninst's DWARF parsing is not accurate for the new ARM machine, which has a very new libc and compiler, I guess that latest compiler is generating DWARF in a way that SymtabAPI does not expect.

**Comment by mxz297** [comment]
@hainest Why is this issue reopened? I thought the new issue is about separate DWARF file.

**Comment by hainest** [comment]
@mxz297 This was closed prematurely- the corresponding PRs haven't been tested/merged yet.

**Comment by sashanicolas** [comment]
The PRs linked here are to fix this issue. They were fixed in one platform and very likely to work on the others. Remaining the test phase on them. That's why I closed this issue. 
Let's keep it simple. There's no need for this issue to be reopened besides to clutter up our issues page @hainest . If the fixes (unlikely) break in another platform, then reopen. Meanwhile we should test the PRs on the other platforms.
